src_kt/java/arcs/crdt/CrdtCount.kt

```diff
@@ -0,0 +1,140 @@
+package arcs.crdt
+
+import arcs.crdt.internal.Actor
+import arcs.crdt.internal.VersionMap
+
+/** A [CrdtModel] capable of managing an increasing [Int] value. */
+class CrdtCount : CrdtModel<CrdtCount.Data, CrdtCount.Operation, Int> {
+  private var _data: Data = Data()
+  override val data: Data
+    get() = _data.copy(values = HashMap(_data.values), versionMap = VersionMap(_data.versionMap))
+
+  override val consumerView: Int
+    get() = _data.values.values.sum()
+
+  fun forActor(actor: Actor) = Scoped(actor)
+
+  inner class Scoped(private val actor: Actor) {
+    private var currentVersion = _data.versionMap[actor] ?: 0
+    private var nextVersion = currentVersion + 1
+
+    fun withCurrentVersion(version: Int) = apply { nextVersion = version }
+    fun withNextVersion(version: Int) = apply { nextVersion = version }
+
+    operator fun plusAssign(delta: Int) {
+      applyOperation(
+        Operation.MultiIncrement(actor, currentVersion to nextVersion, delta)
+      )
+    }
+  }
+
+  override fun merge(other: Data): MergeChanges<Data, Operation> {
+    val myChanges = CrdtChange.Operations<Data, Operation>()
+    val otherChanges = CrdtChange.Operations<Data, Operation>()
+
+    other.values.forEach { (actor, otherValue) ->
+      val myValue = _data.values[actor] ?: 0
+      val myVersion = _data.versionMap[actor] ?: 0
```

Comment:
Optional, but I'd recommend making VersionMap a proper class, which handles creating and returning defaults for you. There's eventually going to be other more complex operations done with VersionMaps (see `mergeVersions`, `sameVersions`, `dominates` functions in the TypeScript implementation) which would make sense to define as methods on VersionMap. Might as well bite that bullet now. Then you won't have to do the `?: 0` dance everywhere.

Reply:
Good thought. I'll do that in the next PR